tor/net.go

```diff
@@ -3,8 +3,11 @@ package tor
 import (
 	"context"
 	"errors"
+	"fmt"
```

Comment:
The `tor` package was turned into a golang Submodule since you opened the PR. This means you need to create a PR for the changes just to the `tor` package that can be merged and tagged and then an additional PR that makes use of the new functionality. The second PR can be created at the same time and use a temporary `replace` directive until the first one is merged. I'm sorry about that more complex process, hopefully the new [workspaces](https://go.dev/ref/mod#workspaces) introduced with Go 1.18 will make this less painful.

Reply:
Cheers for the heads-up. Considering I already did some rebasing work on this branch related to the other recent changes in the module and that this one has already been marked as "milestone v0.16.0" (meaning acting on it now won't make it to 0.15 anyway), I'll return to this once 0.15 is released.

If there's an indication that it can still make the cut for 0.15, happy to get to it sooner tho

Reply:
@guggero been dropping the ball here but would one of these days be a good time to revisit this?

Reply:
Sure, this sounds like something we could include in 0.17. We've started a new parallel development branching model, so if you want, you can open the rebased PRs against the `0-17-0-staging` branch, so we can merge it before 0.16.0 or 0.16.1 is out.